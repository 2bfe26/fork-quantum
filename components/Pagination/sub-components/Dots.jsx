import React from 'react';
import styled from 'styled-components';
import Colors from '../../Colors';

const sharedStyles = `
  background-color: ${Colors.BLACK['200']};
  border-radius: 50%;
  display: inline-block;
  height: 5px;
  width: 5px;
`;

const sharedPseudoElementStyles = `
  content: '';
  position: absolute;
  tpo: 0;
  ${sharedStyles}
`;

const Wrapper = styled.div`
  display: inline-block;
  margin: 0 8px;
  width: 22px;
`;

const Dot = styled.span`
  position: relative;
  ${sharedStyles}

  &:after {
    left: 8px;
    ${sharedPseudoElementStyles}
  }

  &:before {
    left: 16px;
    ${sharedPseudoElementStyles}
  }
`;

const Dots = ({ ...props }) => (
  <Wrapper {...props}>
    <Dot />
  </Wrapper>
);

export default Dots;
