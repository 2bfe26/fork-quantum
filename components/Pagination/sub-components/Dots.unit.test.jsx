import React from 'react';
import { render } from '@testing-library/react';
import Dots from './Dots';

describe('<Dots />', () => {
  describe('Snapshot', () => {
    it('should match snapshot', () => {
      expect(render(<Dots />).asFragment()).toMatchSnapshot();
    });
  });
});
