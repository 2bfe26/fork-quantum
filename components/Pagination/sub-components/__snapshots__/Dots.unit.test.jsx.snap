// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dots /> Snapshot should match snapshot 1`] = `
.c0 {
  display: inline-block;
  margin: 0 4px;
  min-width: 22px;
}

.c1 {
  position: relative;
  background-color: #cccccc;
  border-radius: 50%;
  display: inline-block;
  height: 5px;
  width: 5px;
}

.c1:after {
  left: 8px;
  content: '';
  position: absolute;
  tpo: 0;
  background-color: #cccccc;
  border-radius: 50%;
  display: inline-block;
  height: 5px;
  width: 5px;
}

.c1:before {
  left: 16px;
  content: '';
  position: absolute;
  tpo: 0;
  background-color: #cccccc;
  border-radius: 50%;
  display: inline-block;
  height: 5px;
  width: 5px;
}

<div
  className="c0"
>
  <span
    className="c1"
  />
</div>
`;
